packages/perseus/src/perseus-types.ts

```diff
@@ -11,72 +11,83 @@ export type Size = [number, number];
 export type CollinearTuple = [vec.Vector2, vec.Vector2];
 export type ShowSolutions = "all" | "selected" | "none";
 
+/**
+ * Our core set of Perseus widgets.
+ *
+ * This interface is the basis for "registering" all Perseus widget types.
+ * There should be one key/value pair for each supported widget. If you create
+ * a new widget, an entry should be added to this interface.
+ *
+ * Importantly, the key is not important and is not used anywhere outside of
+ * this interface so it is arbitrary.
+ *
+ * If you define the widget outside of this package, you can still add the new
+ * widget to this interface by writing the following in that pacakge that
+ * contains the widget. TypeScript will merge that definition of the
+ * `PerseusWidgets` with the one defined below.
+ *
+ * ```typescript
+ * declare module "@khanacademy/perseus" {
+ *     interface PerseusWidgetTypes {
+ *         Awesomeness: MyAwesomeNewWidget;
+ *     }
+ * }
+ *
+ * type MyAwesomeNewWidget = WidgetOptions<'awesomeness', MyAwesomeNewWidgetOptions>;
+ * ```
+ *
+ * This interface can be extended through the magic of TypeScript "Declaration
+ * merging". Specifically, we augment this module and extend this interface.
+ *
+ * @see {@link https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation}
```

Comment:
Thanks for adding this docstring.  I think it might also be worth mentioning that `MyAwesomeNewWidget` needs to be registered and what that looks like, but that might make more sense in the README.md.

Reply:
I added a short note about registering widgets in this comment.